src/main/kotlin/graphics/scenery/primitives/Cylinder.kt

```diff
@@ -18,63 +18,95 @@ import kotlin.math.sqrt
  * @param[segments] Number of segments in latitude and longitude.
  */
 
-class Cylinder(var radius: Float, var height: Float, var segments: Int) : Mesh("cylinder") {
+class Cylinder(var radius: Float, var height: Float, var segments: Int, fillCaps: Boolean = false, smoothSides: Boolean = false) : Mesh("cylinder") {
     init {
         geometry {
-            geometryType = GeometryType.TRIANGLE_STRIP
+            geometryType = GeometryType.TRIANGLES
```

Comment:
Triangle strips are more efficient to render (cache efficiency and so on). The stripification could be done as a postprocessing step, so consider this only a note ^^